algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -578,8 +652,19 @@ case class CMSInstance[K](countsTable: CMSInstance.CountsTable[K],
     }
   }
 
-  def frequency(item: K): Approximate[Long] =
-    makeApprox(countsTable.counts.iterator.zip(params.hashes.iterator).map { case (row, hash) => row(hash(item)) }.min)
+  def frequency(item: K): Approximate[Long] = {
+    var freq = Long.MaxValue
+    val hs = params.hashes
+    val it = countsTable.counts.iterator
+    var i = 0
+    while (it.hasNext) {
```

Comment:
I'll try this out -- we should have some benchmarks that test `frequency` anyway.